src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelParameters.java

```diff
@@ -0,0 +1,306 @@
+
+        /* write targets list */
+        final File targetListFile = new File(outputPath, CoverageModelGlobalConstants.TARGET_LIST_OUTPUT_FILE);
+        TargetWriter.writeTargetsToFile(targetListFile, model.getTargetList());
+
+        final List<String> targetNames = model.getTargetList().stream()
+                .map(Target::getName).collect(Collectors.toList());
+
+        /* write target mean bias to file */
+        final File targetMeanBiasFile = new File(outputPath, CoverageModelGlobalConstants.TARGET_MEAN_LOG_BIAS_OUTPUT_FILE);
+        Nd4jIOUtils.writeNDArrayToTextFile(model.getTargetMeanLogBias(), targetMeanBiasFile, null, targetNames);
+
+        /* write target unexplained variance to file */
+        final File targetUnexplainedVarianceFile = new File(outputPath, CoverageModelGlobalConstants.TARGET_UNEXPLAINED_VARIANCE_OUTPUT_FILE);
+        Nd4jIOUtils.writeNDArrayToTextFile(model.getTargetUnexplainedVariance(), targetUnexplainedVarianceFile,
+                null, targetNames);
+
+        /* write bias covariates to file */
+        final List<String> biasCovariatesNames = IntStream.range(0, model.getNumLatents())
+                .mapToObj(li -> String.format("BC_%d", li)).collect(Collectors.toList());
+        final File biasCovariatesFile = new File(outputPath, CoverageModelGlobalConstants.BIAS_COVARIATES_OUTPUT_FILE);
+        Nd4jIOUtils.writeNDArrayToTextFile(model.getBiasCovariates(), biasCovariatesFile, targetNames, biasCovariatesNames);
+
+        /* write norm_2 of bias covariates to file */
+        final double[] biasCovariatesNorm2 = new double[model.numLatents];
+        final INDArray WTW = model.getBiasCovariates().transpose().mmul(model.getBiasCovariates());
+        for (int li = 0; li < model.getNumLatents(); li++) {
+            biasCovariatesNorm2[li] = WTW.getDouble(li, li);
+        }
+        final File biasCovariatesNorm2File = new File(outputPath, CoverageModelGlobalConstants.BIAS_COVARIATES_NORM2_OUTPUT_FILE);
+        Nd4jIOUtils.writeNDArrayToTextFile(Nd4j.create(biasCovariatesNorm2, new int[] {1, model.getNumLatents()}),
+                biasCovariatesNorm2File, Collections.singletonList("NORM_2"), biasCovariatesNames);
+    }
+
+    /**
+     * This method "adapts" a model to a read count collection in the following sense:
+     *
+     *     - removes targets that are not included in the model from the read counts collection
+     *     - removes targets that are in the read count collection from the model
+     *     - rearranges model targets in the same order as read count collection targets
+     *
+     * The modifications are not done in-plane and the original input parameters remain intact.
+     *
+     * @param model a model
+     * @param readCounts a read count collection
+     * @return a pair of model and read count collection
+     */
+    public static ImmutablePair<CoverageModelParameters, ReadCountCollection> adaptModelToReadCountCollection(
```

Comment:
I agree. I made unit test placeholders, TODO, and javadocs to tests to be written.